### PR TITLE
Fractal node state

### PR DESCRIPTION
**Pull request by adamgall**
## Description

Second in a series of PRs, with the overall goal being to allow the webapp to load up either an "MVD"-style Node, or a "Gnosis Safe"-style Node.

## Notes

This single commit stands alone as it's own PR, because I kindly request thoughtful feedback on it. This is the commit where I introduce the first other piece of top-level state being returned from `useFractal`, which is an object called `node`.

It has its own interface, its own actions, its own initial state, _and_ its own `dispatch` function. That's the part where I'm sure there's a better way to do it, but I'm not sure what that better way is with React conventions. It seems dirty to have multiple `dispatch` functions though.

Notice how in `DAOController`, when the MVD DAO is being "set", I call both of the `dispatch` methods. When a DAO is being "unloaded", I call the `RESET` action on both `dispatch` functions. This feels wrong. It works (both pieces of top level state are updated correctly and reset correctly), but I'm sure there's a cleaner way to do this.

Similar to #522, there are no user-facing or functional changes in this PR. But, logic does change here, with the addition of this new piece of top level `node` state.

## Issue (if applicable)

n/a

## Testing

n/a

## Screenshots (if applicable)

n/a

### CODE REVIEW

**Review by herbig — COMMENTED** [review]
a minor comment on the action name, and I'll let the React pros comment on patterns, though you've given me some reading to check out reducers

File: src/providers/fractal/constants/enums.ts [hunk 1]
```diff
@@ -1,3 +1,13 @@
+export enum NodeType {
+  MVD,
+}
+
+export enum NodeAction {
+  SET_NODE_TYPE,
+  INVALID,
```
[the remark this answers is not in the document]
Comment (adamgall) [reply src/providers/fractal/constants/enums.ts:7]: Fixed in latest commit

File: src/providers/fractal/types/node.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { NodeType, NodeAction } from '../constants/enums';
+
+export interface FractalNode {
+  nodeType?: NodeType;
+  isLoading?: boolean;
```
[the remark this answers is not in the document]
Comment (adamgall) [reply src/providers/fractal/types/node.ts:5]: I don't understand what you're asking here @mudrila

File: src/providers/fractal/constants/enums.ts [hunk 3]
```diff
@@ -1,3 +1,13 @@
+export enum NodeType {
+  MVD,
```
[the remark this answers is not in the document]
Comment (adamgall) [reply src/providers/fractal/constants/enums.ts:2]: I could, sure, but it's not being actively _used_ in this PR. I opted to both the new `NodeType` (`GNOSIS`) and _start using it_ in the next PR.

File: src/controller/DAOs/DAOController.tsx [hunk 4]
```diff
@@ -95,29 +96,31 @@ export function DAOController({ children }: { children: JSX.Element }) {
...
 
   useEffect(() => {
     if (!isProviderLoading && (errorMessage || !account)) {
       toast(errorMessage);
-      (async () => {
-        dispatch({
-          type: MVDAction.INVALID,
-        });
-      })();
+      nodeDispatch({ type: NodeAction.INVALIDATE });
```
Comment (herbig) [inline src/controller/DAOs/DAOController.tsx:114]: does removing the `async` from these dispatch calls have any impact for this?

File: src/providers/fractal/constants/state.ts [hunk 5]
```diff
@@ -1,4 +1,9 @@
-import { MVDDAO } from '../types';
+import { MVDDAO, FractalNode } from '../types';
+
+export const nodeInitialState: FractalNode = {
+  nodeType: undefined,
+  isLoading: true,
```
[the remark this answers is not in the document]
Comment (herbig) [reply src/providers/fractal/constants/state.ts:5]: if `nodeType: undefined` indicates nothing has loaded, maybe we can get rid of `isLoading` altogether?

[the remark this answers is not in the document]
Comment (adamgall) [reply src/providers/fractal/constants/state.ts]: Easy fix was to rename the property from `isLoading` to `isLoaded` and negate the logic.

[the remark this answers is not in the document]
Comment (adamgall) [reply src/providers/fractal/constants/state.ts]: > if `nodeType: undefined` indicates nothing has loaded, maybe we can get rid of `isLoading` altogether?

Ah that's a good point. It's not being used, just set right now. Makes sense to just remove it.

[the remark this answers is not in the document]
Comment (adamgall) [reply src/providers/fractal/constants/state.ts]: It's been removed in the latest commit

File: src/controller/DAOs/DAOController.tsx (hunk 4 above)
Comment (adamgall) [reply src/controller/DAOs/DAOController.tsx:114]: It shouldn't, none of the functions inside of the previously-marked `async` anonymous function are `promise`s. It wasn't harming anything, but it was unnecessary.

**Review by herbig — APPROVED** [review]
PR 2 looks good to me

**Pull request merged**